optaplanner-core/src/test/java/org/optaplanner/core/impl/exhaustivesearch/DefaultExhaustiveSearchPhaseTest.java

```diff
@@ -100,19 +100,19 @@ public void restoreWorkingSolution() {
         phase.restoreWorkingSolution(stepScope);
 
         verify(node0.getMove(), times(0)).doMove(any(ScoreDirector.class));
-        verify(node0.getUndoMove(), times(0)).doMove(any(ScoreDirector.class));
+        verify(node0.getUndoMove(), times(0)).doMoveOnly(any(ScoreDirector.class));
         verify(node1.getMove(), times(0)).doMove(any(ScoreDirector.class));
-        verify(node1.getUndoMove(), times(0)).doMove(any(ScoreDirector.class));
+        verify(node1.getUndoMove(), times(0)).doMoveOnly(any(ScoreDirector.class));
         verify(node2A.getMove(), times(0)).doMove(any(ScoreDirector.class));
-        verify(node2A.getUndoMove(), times(1)).doMove(scoreDirector);
+        verify(node2A.getUndoMove(), times(1)).doMoveOnly(scoreDirector);
         verify(node3A.getMove(), times(0)).doMove(any(ScoreDirector.class));
-        verify(node3A.getUndoMove(), times(1)).doMove(scoreDirector);
-        verify(node2B.getMove(), times(1)).doMove(scoreDirector);
-        verify(node2B.getUndoMove(), times(0)).doMove(any(ScoreDirector.class));
-        verify(node3B.getMove(), times(1)).doMove(scoreDirector);
-        verify(node3B.getUndoMove(), times(0)).doMove(any(ScoreDirector.class));
-        verify(node4B.getMove(), times(1)).doMove(scoreDirector);
-        verify(node4B.getUndoMove(), times(0)).doMove(any(ScoreDirector.class));
+        verify(node3A.getUndoMove(), times(1)).doMoveOnly(scoreDirector);
+        verify(node2B.getMove(), times(1)).doMoveOnly(scoreDirector);
+        verify(node2B.getUndoMove(), times(0)).doMoveOnly(any(ScoreDirector.class));
+        verify(node3B.getMove(), times(1)).doMoveOnly(scoreDirector);
+        verify(node3B.getUndoMove(), times(0)).doMoveOnly(any(ScoreDirector.class));
+        verify(node4B.getMove(), times(1)).doMoveOnly(scoreDirector);
+        verify(node4B.getUndoMove(), times(0)).doMoveOnly(any(ScoreDirector.class));
```

Comment:
The test passes like this, but I do not really understand why it needs to be like that.
The test is somewhat cryptic.

Reply:
IIRC, it's the delta set of moves to go from one solution state to a totally different one, somewhere in another node in the big exponential search tree.